app/src/main/java/ch/epfl/skysync/screens/reports/ReportDetailsScreen.kt

```diff
@@ -37,8 +38,10 @@ fun ReportDetailsScreen(
     val display = (isAdmin && reportIds!!.isNotEmpty()) || reportIds!!.any { (it.author == userId) }
     Column() {
       CustomTopAppBar(navController, "Report")
+      reportIds!!.forEach { report -> Log.d("report", report.author) }
       if (display) {
-        ReportDetail(users, reportIds, isAdmin, userId)
+        val reportLists = if (isAdmin) reportIds else reportIds!!.filter { (it.author == userId) }
```

Comment:
same as above. Ideally, the VM (which know the user) will return the relevant reportIds

Reply:
Big iq comment ty